docs/intro/developing.rst

```diff
@@ -7,19 +7,33 @@ Developing New Components
 The *dwave-hybrid* framework enables you to build your own components to incorporate into your
 workflow.
 
-The key superclass is the :class:`~hybrid.core.Runnable` class: the basic components---samplers,
-decomposers, composers---and the flow-structuring components such as branches inherit
+The key superclass is the :class:`~hybrid.core.Runnable` class: all basic components---samplers,
+decomposers, composers---and flow-structuring components such as branches inherit
 from this class. A :class:`~hybrid.core.Runnable` is run for an iteration in which it updates
 the :class:`~hybrid.core.State` it receives. Typical methods are `run` or `next` to execute an
 iteration and `stop` to terminate the :class:`~hybrid.core.Runnable`.
 
-The :ref:`core` and :ref:`flow` sections describe, respectively the basic :class:`~hybrid.core.Runnable`
+The :ref:`core` and :ref:`flow` sections describe, respectively, the basic :class:`~hybrid.core.Runnable`
 classes (building blocks) and flow-structuring ones and their methods. If you are
 implementing these methods for your own :class:`~hybrid.core.Runnable` class, see comments in
 the code.
 
+The :ref:`racingBranches1` graphic below shows the top-down composition (tree structure) of a hybrid loop.
+
+.. figure:: ../_static/tree.png
+  :name: Tree
+  :scale: 90 %
+  :alt: Top-Down Composition
+
+  Top-Down Composition
+
+Your code should enforce trait verification on :class:`~hybrid.core.State` objects, at a minimum
+verify correct inputs and outputs (full verification on some branches may be overly complex).
+Note that if you define an :code:`__init__` method, you must inherit from the
+:class:`~hybrid.traits.StateTraits` class to implement traits.
```

Comment:
To add traits, users (developers) do not have to inherit from `StateTraits` directly - all classes (*traits*) in the `hybrid.traits` module are subclasses of `StateTraits`, e.g. `ProblemIntaking`, `ProblemDecomposer` and `SIMO` all inherit from the base traits class - so they all add state traits' constraints to a runnable.

Reply:
Thanks for this helpful clarification. How about "Some trait verification is applied to :class:`~hybrid.core.State` objects, at a minimum verifying correct inputs and outputs (full verification on some branches may be overly complex to implement). If you develop advanced flow-control classes, and override an :code:`__init__` method, call the parent class's :code:`__init__` with a `super` construct as done in the subclasses of the :class:`~hybrid.traits.StateTraits` class."?
This is for the 3 comments above

Reply:
What we want to say here is:
- State traits verification if done automatically for all runnables that inherit from `hybrid.traits.StateTraits`, or some of its subclasses.
- There are two levels of that verification: (1) minimal checks at workflow (composition of runnables) construction, and (2) definite checks at run time.
- Constructing composite traits when composing runnables might not be trivial. Parent runnable must express child's traits and modify them at construction time.
- All built-in runnables declare state traits requirements which are either independent (for simple runnables) or derived from the child workflow.
- Custom block developers are encouraged to declare state traits requirements, but they are not required to do so, especially for advanced flow-control runnables.